src/main.rs

```diff
@@ -16,7 +17,23 @@ use wrangler::version::background_check_for_updates;
 use anyhow::Result;
 use structopt::StructOpt;
 
-fn main() -> Result<()> {
+fn main() {
+    if let Err(err) = main_handling_errors() {
+        let mut msg = format!("Error: {}", err);
+        for cause in err.chain().skip(1) {
+            write!(msg, "\n\nCaused by:\n    {}", cause).unwrap();
+        }
+        eprintln!("{}", msg);
+
+        let backtrace = err.backtrace().to_string();
+        if backtrace != "disabled backtrace" && backtrace != "unsupported backtrace" {
+            eprintln!("\nStack backtrace:\n{}", backtrace);
+        }
+        std::process::exit(1);
+    }
+}
```

Comment:
I want to upstream this to rust-lang/rust at some point, but I'm not sure if they'll accept the PR until backtrace is stabilized :/